packages/ui/src/containers/ApplyByRoleContainer/ApplyByRoleContainer.test.tsx

```diff
@@ -1,12 +1,71 @@
-import { render } from "../../../utils/jest-apollo";
+import { getProject } from "@eden/package-mock";
+import userEvent from "@testing-library/user-event";
+import { RouterContext } from "next/dist/shared/lib/router-context";
+
+import { render, screen } from "../../../utils/jest-apollo";
+import { createMockRouter } from "../../../utils/test-utils/createMockRouter";
 import { ApplyByRoleContainer } from "./";
 
-describe("ApplyByRoleContainer", () => {
-  it("renders without throwing", () => {
-    const { container } = render(
-      <ApplyByRoleContainer onViewProject={(val) => console.log(val)} />
-    );
+jest.mock("react-confetti");
+
+test("User can get to the YOU DID IT modal ", async () => {
+  const user = userEvent.setup();
+
+  await render(
+    <RouterContext.Provider value={createMockRouter({})}>
+      <ApplyByRoleContainer
+        onViewProject={(val) => console.log(val)}
+        loadingProject={false}
+        project={getProject()}
+      />
+    </RouterContext.Provider>
+  );
+
+  //There are multiple roles on a page with multiple "more" buttons. Here I am clicking on the first one:
+  await user.click(screen.getAllByText("More")[0]);
+
+  /*
+   By clicking on the "More" button we open up the ApplyByRoleModal in which a user can fill out more information
+
+   --Bellow are the tests for this modal:
+ */
+  //User selects the timezone from the dropdown
+  await user.click(screen.getByRole("combobox"));
+  await user.click(screen.getByText("UTC-10"));
```

Comment:
You might be able to use `role="option"` here?

```suggestion
  await user.click(screen.getByRole("option", { name: "UTC-10" }));
```

Reply:
Yes, that does work! So the `name`, in this case, would be one of the selections in the dropdown, correct?